File: backend/src/main/java/com/ecom/salezone/services/WishlistService.java
package com.ecom.salezone.services;

import com.ecom.salezone.dtos.AddToWishlistRequest;
import com.ecom.salezone.dtos.WishlistDto;

public interface WishlistService {

    WishlistDto addToWishlist(String userId, AddToWishlistRequest request, String logkey);

    WishlistDto getWishlist(String userId, String logkey);

    void removeFromWishlist(String userId, String productId, String logkey);
}

File: frontend/src/services/wishlistService.js
import api from "./api";

const wishlistService = {
  getWishlist: async (userId) => {
    const res = await api.get(`/wishlist/${userId}`);
    return res.data;
  },

  addToWishlist: async (userId, productId) => {
    const res = await api.post(`/wishlist/${userId}`, { productId });
    return res.data;
  },

  removeFromWishlist: async (userId, productId) => {
    const res = await api.delete(`/wishlist/${userId}/${productId}`);
    return res.data;
  },
};

export default wishlistService;
